src/ckeditor.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* global CKEDITOR */
+
+import { getEditorNamespace } from './helpers.js';
+
+export default {
+	name: 'ckeditor',
+
+	render( createElement ) {
+		return createElement( 'div', {}, [
+			createElement( this.tagName )
+		] );
+	},
+
+	props: {
+		value: {
+			type: String,
+			default: ''
+		},
+		type: {
+			type: String,
+			default: 'classic',
+			validator: type => type in { classic: 1, inline: 1 }
+		},
+		editorUrl: {
+			type: String,
+			default: 'https://cdn.ckeditor.com/4.11.4/standard/ckeditor.js'
+		},
+		config: {
+			type: Object,
+			default: () => {}
+		},
+		tagName: {
+			type: String,
+			default: 'textarea'
+		},
+		readOnly: {
+			type: Boolean,
+			default: null // Use the null for default value, so `config.readOnly` can take precedence.
+		}
+	},
+
+	readOnly() {
+		return this.readOnly === null
+			? this.instance.readOnly
+			: this.readOnly;
+	},
+
+	beforeCreate() {
+		this.$_ready = new Promise( res => {
+			this.$once( 'ready', res );
+		} );
+	},
+
+	mounted() {
+		getEditorNamespace( this.editorUrl ).then( () => {
+			const editor = this.instance =
+				CKEDITOR[ this.type === 'inline' ? 'inline' : 'replace' ]( this.$el.firstElementChild, this.config );
+
+			editor.on( 'instanceReady', () => {
+				if ( this.readOnly !== null ) {
+					this.instance.setReadOnly( this.readOnly );
+				}
+
+				const undo = editor.undoManager;
+				const data = this.value;
+
+				if ( data !== null ) {
+					undo && undo.lock();
+
+					editor.once( 'dataReady', () => {
+						this.$_setUpEditorEvents();
+
+						// Locking undoManager prevents 'change' event.
+						// Trigger it manually to update bound data.
+						if ( data !== editor.getData() ) {
+							editor.once( 'change', () => {
+								this.$emit( 'ready', editor );
+							} );
+
+							editor.fire( 'change' );
+						} else {
+							this.$emit( 'ready', editor );
+						}
+
+						undo && undo.unlock();
+					} );
+
+					editor.setData( data );
+				} else {
+					this.$_setUpEditorEvents();
+					this.$emit( 'ready', editor );
+				}
+			} );
+		} );
+	},
+
+	beforeDestroy() {
+		this.destroyEditor();
+
+		// Note: By the time the editor is destroyed (editor.on( 'destroy' ) called)
+		// the Vue component will not be able to emit any longer. So emitting #destroy a bit earlier.
+		this.$emit( 'destroy', this.instance );
+	},
+
+	watch: {
+		value( val ) {
+			if ( this.instance && this.instance.getData() !== val ) {
```

Comment:
`readOnly` should be validated too. It can be updated before instance is created, it would lead to an error.

Reply:
It seems for me like developer error than an implementation issue. What's a use case for setting data / readonly mode before the editor is initialized? If it can be avoided by developer e.g. by listening on `ready` event, it should be his responsibility to handle this case. Otherwise, data loss will occur, because you are not handling lazy loading here.

If the issue could be caused by framework itself, I would add some caching so data/readonly mode are set despite bad timing.